File: app/src/main/java/com/github/mrvilkaman/presentationlayer/fragments/testfrags/Frag3ScreenFragment.java
package com.github.mrvilkaman.presentationlayer.fragments.testfrags;

import com.github.mrvilkaman.di.ActivityCoreComponent;
import com.github.mrvilkaman.presentationlayer.fragments.core.BaseFragment;

public class Frag3ScreenFragment extends FragBaseScreenFragment {


	public static Frag3ScreenFragment open() {
		return new Frag3ScreenFragment();
	}

	@Override
	protected int getNumber() {
		return 3;
	}

	@Override
	protected BaseFragment nextFragment() {
		return Frag4ScreenFragment.open();
	}

	@Override
	protected int getIcon() {
		return android.R.drawable.ic_delete;
	}

	@Override
	public void daggerInject() {
		ActivityCoreComponent component = getComponent(ActivityCoreComponent.class);
		DaggerFragScreenComponent.builder().activityCoreComponent(component).build().inject(this);
	}
}

File: kotlinapp/src/main/java/com/github/mrvilkaman/ui/screens/testfrags/Frag3ScreenFragment.kt
package com.github.mrvilkaman.ui.screens.testfrags

import com.github.mrvilkaman.R
import com.github.mrvilkaman.di.ActivityComponent
import com.github.mrvilkaman.ui.screens.ScreenKey

class Frag3ScreenFragment : FragBaseScreenFragment() {

    override val number: Int
        get() = 3

    override val icon: Int
        get() = android.R.drawable.ic_delete

    override fun nextScreenKey(): String? = ScreenKey.FRAG4

    override fun daggerInject() {
        val component = getComponent(ActivityComponent::class.java)
        DaggerFragScreenComponent.builder().activityComponent(component).build().inject(this)
    }

    override fun onBackPressed(): Boolean {
        uiResolver.showToast(R.string.cleanbase_simple_text,"QWETR")
        return super.onBackPressed()
    }

    companion object {


        fun open(): Frag3ScreenFragment {
            return Frag3ScreenFragment()
        }
    }
}
